scripts/get_gh_stats.R

```diff
@@ -0,0 +1,296 @@
+        )
+}
+
+#' Get repo
+#'
+#' Get information for a repository
+#'
+#' @param repo Name of the repository (user/repo)
+#'
+#' @return list containing repository information
+get_repo <- function(repo) {
+    message("Getting information for ", repo, "...")
+    endpoint <- glue("GET /repos/{repo}")
+    issues <- gh(endpoint)
+    return(issues)
+}
+
+#' Get repo data frame
+#'
+#' Get information for a repository as a data frame
+#'
+#' @param repo Name of the repository (user/repo)
+#'
+#' @return tibble containing repository information
+get_repo_df <- function(repo) {
+    info <- get_repo(repo)
+    tibble(
+        Repo    = repo,
+        Created = info$created_at,
+        Updated = info$updated_at,
+        Stars   = info$stargazers_count,
+        Forks   = info$forks_count
+    ) %>%
+        mutate(
+            across(c("Created", "Updated"), as_datetime)
+        ) %>%
+        mutate(AgeYears = as.numeric((now() - Created) / 365))
+}
+
+# MAIN CODE --------------------------------------------------------------------
+
+repos <- c(
+    "Teichlab/bbknn",
+    "hms-dbmi/conos",
+    "immunogenomics/harmony",
+    "MacoskoLab/liger",
+    "chriscainx/mnnpy",
+    "brianhie/scanorama",
+    "YosefLab/scVI",
+    "satijalab/seurat",
+    "theislab/trVAE",
+    "jtleek/sva-devel",
+    "eleozzr/desc",
+    "KrishnaswamyLab/SAUCIE",
+    "LTLA/batchelor",
+    "theislab/scgen"
+)
+
+# Get the combined issues data frame for all repos
+all_issues <- map_dfr(repos, get_issues_df)
+
+# Summarise issues by repository
+issues_summ <- all_issues %>%
+    mutate(
+        TimeResponse = Response - Created,
+        TimeClosed   = Closed - Created
+    ) %>%
+    group_by(Repo) %>%
+    summarise(
+        Issues             = n(),
+        Closed             = sum(State == "closed"),
+        PctClosed          = Closed / Issues * 100,
+        MedianResponseDays = as.numeric(median(TimeResponse, na.rm = TRUE)) /
+                                 86400,
+        MedianOpenDays     = as.numeric(median(TimeClosed, na.rm = TRUE)) /
+                                 86400,
+        .groups            = "drop"
+    )
+
+# Get combined data frame with all repo information
+all_repos <- map_dfr(repos, get_repo_df)
+
+# Calculate score for each repository
+repo_summ <- all_repos %>%
+    left_join(issues_summ, by = "Repo") %>%
+    mutate(Tool = str_remove(Repo, "^[[:alnum:]-_]+/")) %>%
+    relocate(Repo, Tool) %>%
+    mutate(
+        # Add 1 to avoid negative scores
+        IssueActivity = log10((Closed / AgeYears) + 1),
+        IssueResponse = 30 - MedianResponseDays
+    ) %>%
+    mutate(
+        IssueActivity = if_else(
+            is.na(IssueActivity),
+            0.5 * min(IssueActivity, na.rm = TRUE),
+            IssueActivity
+        ),
+        IssueResponse = if_else(
+            is.na(IssueResponse),
+            0.5 * min(IssueResponse, na.rm = TRUE),
+            IssueResponse
```

Comment:
is this still needed? If not, I would probably just equate it to the min rather than 0.5*min(). That feels a bit harsh.

Reply:
Probably not needed now for these repos. I had it as `0.5 * min()` so that a repo with some responses would always get a better score than one with no responses.

Reply:
Fair point... `0.5*min()` just seems a bit arbitrary. But never mind then. If it's not used anymore, then leave it... in case someone else wants to use it.